grazer/src/main/java/edu/asu/diging/grazer/core/graphs/db/impl/GraphDBConnection.java

```diff
@@ -39,7 +39,14 @@ public void store(Graph graph) {
     public List<Graph> getGraphs(String conceptUri) {
         Query query = sessionFactory.getCurrentSession().createQuery("SELECT g from Graph g WHERE g.conceptUri = :uri");
         query.setParameter("uri", conceptUri);
-        return query.list();
+        List<Graph> results = query.list();
+        
+        if(results.isEmpty()) {
+            query = sessionFactory.getCurrentSession().createQuery("SELECT g from Graph g JOIN g.nodes n WHERE n.uri = :uri");
+            query.setParameter("uri", conceptUri);
+            results = query.list();
+        }
```

Comment:
- line 40 is unneccessary since line 45 should achieve the same thing
- this still requires you to filter edges afterwards, just get only the edges from the db that contain the url and return them (it might mean you have to refactor the code to work with edges rather than graphs).